### DISCUSSION TITLE
[Usage] How to pass an inline argument for singularity shell call?

### DISCUSSION THREAD

**Discussion by physicistsam** [sylabs/singularity#2748] [Q&A] [answered]
Its a question, rather than an issue.
Is there any way to pass an inline argument for singularity shell call
```
#! /bin/bash
# following arguments are the arguments for the script
filename=$1 
ptg=$2 
data_block=$3
locno=$4


# Path to your Singularity container
SINGULARITY_CONTAINER=/Path-to-your /Singularity /container.sif
# Command to run an interactive shell session with Singularity
singularity shell $SINGULARITY_CONTAINER << 'EOF'
  # some other stuffs
  # Run the script
  ./my_script.sh $filename $ptg $data_block $locno
  
EOF
```
This does not pass the arguments <filename, ptg, data_block, locno>.

**Comment by dtrudg** [comment, marked as answer]
This is a shell scripting issue, relating to the heredoc.

When you use a heredoc `<< 'EOF'` then the quotes (or lack of) around the `EOF` bit control whether variable names will be subsituted.

Single quotes... value of FOO doesn't get to the container shell.

```
$ cat script.sh 
#!/bin/bash

FOO=BAR

singularity shell docker://alpine << 'EOF'
  echo FOO IS: $FOO
EOF

$ bash script.sh 
INFO:    Using cached SIF image
FOO IS:
```

No quotes around EOF... value of FOO does get to the container shell.

```
$ cat script.sh 
#!/bin/bash

FOO=BAR

singularity shell docker://alpine << EOF
  echo FOO IS: $FOO
EOF

$ bash script.sh 
INFO:    Using cached SIF image
FOO IS: BAR
```

See e.g. https://phoenixnap.com/kb/bash-heredoc for more examples.

**Reply by physicistsam** [reply]
It works ! Thanks !
